### ISSUE TITLE
(*LineTable).funcName() causes panic: runtime error: slice bounds out of range [795306342:43562440]

### ISSUE THREAD

**Issue by keisku** [go-delve/delve#3650]
Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

https://github.com/go-delve/delve/pull/3549 seems to introduce this panic because `(*LineTable).funcName()` didn't exist at v1.21.0 source code.

See https://gist.github.com/keisku/a217ce1d3772617c48b3fd1bbcfe03fe for more detail of this issue.

If this is about source listing not showing up (or breakpoints not being
accepted) while running in a container please read our
[FAQ](https://github.com/go-delve/delve/blob/master/Documentation/faq.md#substpath)
first.

1. What version of Delve are you using (`dlv version`)? 

```
Delve Debugger
Version: 1.22.0
Build: efaf5523d8425267f379f4efab8e3144620a0a40
```

2. What version of Go are you using? (`go version`)?

```
go version
go version go1.21.6 linux/amd64
```

3. What operating system and processor architecture are you using?

```
uname -a
Linux ip-10-0-133-150 6.2.0-1017-aws #17~22.04.1-Ubuntu SMP Fri Nov 17 21:07:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

4. What did you do? See the gist above.
5. What did you expect to see? No panic as dlv v1.21.0
6. What did you see instead?

**Comment by aarzilli** [comment]
@derekparker I think this might be a problem with gore.

**Comment by derekparker** [comment]
I will start taking a look at this issue today.

**Comment by keisku** [comment]
FYI:

When this commit is head, this issue caused.

```bash
$ git switch --detach f32818c9e36e6c6e84a07131c08f994ee38fa156
Previous HEAD position was 4a72c523 pkg/proc: remove usage of gore dependency (#3664)
HEAD is now at f32818c9 cmd/dlv: fix --continue with unix domain sockets (#3658)
```

Since https://github.com/go-delve/delve/pull/3664 was merged, this issue have not been caused.

```bash
git switch --detach 4a72c523bed853ceed7bee6272d4c84641fc3b1e
Previous HEAD position was f32818c9 cmd/dlv: fix --continue with unix domain sockets (#3658)
HEAD is now at 4a72c523 pkg/proc: remove usage of gore dependency (#3664)
```
